recipes/flyteidl/meta.yaml

```diff
@@ -0,0 +1,45 @@
+{% set name = "flyteidl" %}
+{% set version = "0.21.24" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/flyteidl-{{ version }}.tar.gz
+  sha256: e4c992f646744ea34066e3f4154601f891f7a9cfbcaf90228fd19e1dfd8dab21
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - protobuf >=3.5.0,<4.0.0
+    - python
+    - typing >=3.6  # [py==27]
```

Comment:
Sorry, missed that the first time. Similarly to https://github.com/conda-forge/staged-recipes/pull/17778#discussion_r797990317 you can remove this as conda-forge cannot build `python2.7`, make the package `noarch` and restrict the lowest python version to whatever upstream lists as the lowest version that is larger than `2.7`.

Reply:
OK, will do. Couple questions though:

1. By upstream, do you mean `flytekit`? This is the main package of interest for us. I don't think it directly requires a specific version of python, unless I'm misinterpreting.
2. Do we even need to bring in `singledispatchmethod`, if we set the python version to `>=3.7`? This would then nullify the need to specify this `typing` version constraint, right?

Reply:
RE 1: I meant `flyteidl`. Basically, you'll need to figure out if they have a minimum python version they support and then add that to the `host` and `run` python, similarly to what you did with `singledispatchmethod`. If they have none, `python>=3.6` is a safe bet.

RE 2: If you set the minimum version of `flytekit` to `python>=3.8` (Not `python>=3.7` since it applies to everything lower than `python3.8`) then no, you wouldn't need `singledispatchmethod`. That is up to you as maintainer of the conda-forge distribution. That would also mean that you can probably switch to `noarch` with that package although I haven't had a look at your windows issue so that is no definitive.

Reply:
On a side-note: On conda-forge `typing` is an empty package as we don't support `python2.7` so it wouldn't do anything anyways.

Ref: https://conda-forge.org/docs/maintainer/knowledge_base.html#empty-python-packages

Reply:
👍 I'd be in favor of just bumping to py 3.8+ and going noarch for the time being. That'll make post-merge dance of bring up builds easier.

Reply:
OK, I'll change `flytekit` to a noarch build and bump the version to 3.8+. Shall I also remove the port in of `singledispatchmethod`?

Reply:
For sure, I'm ready to burn the boats on py38 anyway. 😉

Reply:
OK sweet, same here!